src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -1,38 +1,122 @@
 import java.sql.Types;
+import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 import org.bson.BsonBoolean;
 import org.bson.BsonDocument;
 import org.bson.BsonElement;
-import java.util.ArrayList;
-import java.util.regex.Pattern;
 import org.bson.BsonInt32;
 import org.bson.BsonNull;
 import org.bson.BsonString;
 import org.bson.BsonValue;
+import org.bson.Document;
 
 public class MongoSQLDatabaseMetaData extends MongoDatabaseMetaData implements DatabaseMetaData {
 
     private static final String BOT_NAME = "";
-
-    private static com.mongodb.jdbc.MongoSQLFunctions MongoSQLFunctions =
+    private static final String INDEX_KEY_KEY = "key";
+    private static final String INDEX_NAME_KEY = "name";
+
+    private static final List<String> UNIQUE_KEY_PATH = Arrays.asList("options", "unique");
+
+    private static final List<SortableBsonDocument.SortSpec> GET_TABLES_SORT_SPECS =
+            Arrays.asList(
+                    new SortableBsonDocument.SortSpec(
+                            TABLE_TYPE, SortableBsonDocument.ValueType.String),
+                    new SortableBsonDocument.SortSpec(
+                            TABLE_CAT, SortableBsonDocument.ValueType.String),
+                    new SortableBsonDocument.SortSpec(
+                            TABLE_NAME, SortableBsonDocument.ValueType.String));
+
+    private static final List<SortableBsonDocument.SortSpec> GET_TABLE_PRIVILEGES_SORT_SPECS =
+            Arrays.asList(
+                    new SortableBsonDocument.SortSpec(
+                            TABLE_CAT, SortableBsonDocument.ValueType.String),
+                    new SortableBsonDocument.SortSpec(
+                            TABLE_NAME, SortableBsonDocument.ValueType.String));
+
+    private static final List<SortableBsonDocument.SortSpec> GET_COLUMNS_SORT_SPECS =
+            Arrays.asList(
+                    new SortableBsonDocument.SortSpec(
+                            TABLE_CAT, SortableBsonDocument.ValueType.String),
+                    new SortableBsonDocument.SortSpec(
+                            TABLE_NAME, SortableBsonDocument.ValueType.String),
+                    new SortableBsonDocument.SortSpec(
+                            ORDINAL_POSITION, SortableBsonDocument.ValueType.Int));
+
+    private static final List<SortableBsonDocument.SortSpec> GET_COLUMN_PRIVILEGES_SORT_SPECS =
+            Collections.singletonList(
+                    new SortableBsonDocument.SortSpec(
+                            COLUMN_NAME, SortableBsonDocument.ValueType.String));
+
+    private static final List<SortableBsonDocument.SortSpec> GET_PRIMARY_KEYS_SORT_SPECS =
```

Comment:
This is the same as the one above (`GET_COLUMN_PRIVILEGES_SORT_SPECS`). Should we use the same one twice or construct these two separately so we can stick with the naming pattern?